moha/utils.py

```diff
@@ -4,13 +4,40 @@
 
 from scipy.sparse import diags
 
+import re
 
 __all__ = [
     "convert_indices",
     "get_atom_type",
     "expand_sym",
 ]
+def get_atom_type(atom):
+    """
+    Return atom type and position index, where atom type includes the site index from
+    parentheses appended to the main atom type, e.g., "C2(3)" -> "C3", 2.
 
+    :param atom: str - atom description
+    :return: tuple - (atom type with site, position index)
+    """
+    # The pattern matches an initial letter sequence for the atom type,
+    # followed by a number for the position, and numbers in parentheses for site index to be appended.
+    pattern = r"([A-Za-z]+)(\d+)\((\d+)\)"
+    match = re.match(pattern, atom)
+    
+    if match:
+        # If the pattern matches, append the site index to the atom type
+        atom_type = match.group(1) + match.group(3)
+        position_index = int(match.group(2))
+        return atom_type, position_index
+    else:
+        # Fallback for handling cases without parentheses
+        i = 1
+        while i <= len(atom) and atom[-i].isdigit():
+            i += 1
+        i -= 1
+        if i == 0:
+            raise ValueError(f"Invalid atom format: {atom}")
+        return atom[:-i], int(atom[-i:])
```

Comment:
instead of return statement we can be consistent with the previous case and say `atom_type, position_index = atom[:-i], int(atom[-i:])`